cartography/intel/aws/codeguru.py

```diff
@@ -0,0 +1,249 @@
+    return response['CodeReviewSummaries']
+
+@timeit
+@aws_handle_regions
+def get_recommendations(boto3_session: boto3.session.Session,region: str,code_reviews: List[str]) -> List[str]:
+    client = boto3_session.client('codeguru-reviewer', region_name=region)
+    recommendations: List[Any] = []
+    queues: List[Dict] = []
+    for cr in code_reviews:
+        response = client.list_recommendations(CodeReviewArn=cr['CodeReviewArn'])
+        for queue in response['RecommendationSummaries']:
+            queue['CodeReviewArn'] = cr['CodeReviewArn']
+            queues.append(queue)
+
+    return queues
+
+@timeit
+def load_repository_associations(
+    neo4j_session: neo4j.Session,
+    data: list,
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    cypher = """
+    UNWIND {Repos} AS repo
+        MERGE (cga:CodeguruAssociation{associationarn: repo.AssociationArn})
+        ON CREATE SET cga.firstseen = timestamp()
+        SET  
+             cga.associationid = repo.AssociationId,
+             cga.name = repo.Name,
+             cga.owner = repo.Owner,
+             cga.providertype = repo.ProviderType,
+             cga.state = repo.State,
+             cga.lastupdated = {aws_update_tag}
+        WITH cga
+        MATCH (owner:AWSAccount{id: {AWS_ACCOUNT_ID}})
+        MERGE (owner)-[r:RESOURCE]->(cga)
+        ON CREATE SET cga.firstseen = timestamp()
+        SET cga.lastupdated = {aws_update_tag}
+    """
+
+    neo4j_session.run(
+        cypher,
+        Repos=data,
+        Region=region,
+        AWS_ACCOUNT_ID=current_aws_account_id,
+        aws_update_tag=aws_update_tag,
+    )
+
+@timeit
+def load_code_reviews(
+    neo4j_session: neo4j.Session,
+    data: list,
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    cypher = """
+    UNWIND {Repos} AS repo
+        MERGE (cgc:CodeguruCodereview{codereviewarn: repo.CodeReviewArn})
+        ON CREATE SET cgc.firstseen = timestamp()
+        SET  
+             cgc.associationarn = repo.AssociationArn,
+             cgc.name = repo.Name,
+             cgc.owner = repo.Owner,
+             cgc.providertype = repo.ProviderType,
+             cgc.pullrequestId = repo.PullRequestId,
+             cgc.repositoryname = repo.RepositoryName,
+             cgc.state = repo.State,
+             cgc.type = repo.Type,
+             cgc.lastupdated = {aws_update_tag}
+        WITH cgc,repo
+        MATCH (cga:CodeguruAssociation{associationarn: repo.AssociationArn})
+        MERGE (cga)-[r:HAS_CODEREVIEW]->(cgc)
+        ON CREATE SET cgc.firstseen = timestamp()
+        SET cgc.lastupdated = {aws_update_tag}
+    """
+    queues: List[Dict] = []
+    for queue in data:
+        queue['AssociationArn'] = queue['CodeReviewArn'].split(':code-review:')[0]
+        queue['CodeReviewArn'] = queue['CodeReviewArn']
```

Comment:
I think since you're mutating the data in the original `data` array, you can avoid making the `queues` array and appending to it, and directly pass `data` into `neo4j_session.run`. This will save some memory.